test_unstructured_ingest/dest/pinecone.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+
+if [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because PINECONE_API_KEY env var is not set."
+   exit 0
+fi
+
+RANDOM_SUFFIX=$((RANDOM % 100000 + 1))
```

Comment:
Use this to create a random value:
```shell
RANDOM_SUFFIX=$(uuidgen)
```

Reply:
Such index names did not work, giving a 400 response; example:
`ingest-test-3FE880DE-029B-484F-91FC-417BE200EF03`

Is there any reason we shouldn't use `RANDOM`? If we have to modify it to `uuidgen`, we'll need custom modification in the id after generating it

Reply:
Only because that's what we use everywhere else. If that doesn't work, mind leaving a comment here why we're using a different approach?

Reply:
Yes, added with [add comment on why we use random rather than uuidgen](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/738d75c8faf488b09f4eeddaa36652d2db71db8d)